marklogic-client-api/src/main/java/com/marklogic/client/spark/SparkTask.java

```diff
@@ -0,0 +1,72 @@
+package com.marklogic.client.spark;
+
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.marklogic.client.DatabaseClient;
+import com.marklogic.client.DatabaseClientFactory;
+import com.marklogic.client.dataservices.OutputEndpoint;
+import com.marklogic.client.io.JacksonHandle;
+import com.marklogic.client.kafka.Common;
+import com.marklogic.client.kafka.IOTestUtil;
+import org.apache.spark.sql.SparkSession;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+
+public class SparkTask implements Runnable {
+    private static final Logger logger = LoggerFactory.getLogger(SparkTask.class);
+
+    static ObjectNode apiObj;
+    static String apiName;
+    static String scriptPath;
+    static String apiPath;
+    OutputEndpoint.BulkOutputCaller bulkCaller;
+    DatabaseClient db;
+    int recordCount;
+    int id;
+    SparkSession spark;
+    String productName;
+    DatabaseClient restClient;
+    DatabaseClient noRestClient;
+    String endPointModulePath;
+    String endPointModuleName;
+    String modelPath;
+    String includeTraining;
+
+    public SparkTask(String forestName) {
+        try {
+            apiName = "bulkOutputCallerNext.api";
+            db = DatabaseClientFactory.newClient("localhost", 8012,
+                    new DatabaseClientFactory.DigestAuthContext("admin", "admin"),
+                    Common.CONNECTION_TYPE);
+            apiObj = IOTestUtil.readApi(apiName);
+            scriptPath = IOTestUtil.getScriptPath(apiObj);
+            apiPath = IOTestUtil.getApiPath(scriptPath);
+            IOTestUtil.load(apiName, apiObj, scriptPath, apiPath);
```

Comment:
Instead of repeating the initialization logic for loading the script in each instance of the task for a thread, would it make sense to move it into the Connector?